src/codegen/codegen_c_visitor.cpp

```diff
@@ -2021,6 +2021,19 @@ void CodegenCVisitor::print_nmodl_constants() {
                                           stod(it->get_value()->get_value())));
         }
     }
+    if (!info.constants_definitions.empty()) {
+        printer->add_newline(2);
+        printer->add_line("/** constants used in nmodl from CONSTANTS */");
+        for (const auto& it: info.constants_definitions) {
+            const std::string format_string = "static const {} {{}} = "_format(
+                it->get_value()->get_node_type() == AstNodeType::DOUBLE ? "double" : "int");
+            printer->add_indent();
+            printer->add_text(fmt::format(format_string, it->get_name()->get_value()->get_value()));
+            it->get_value()->accept(*this);
```

Comment:
no nicier way?

Reply:
usually I use the proper visit function instead of accept. However, this works as well. For the printing lines I let @pramodk talk because I always pick the wrong way to print stuff. I would check how the range variables are declared

Reply:
maybe we can use constexpr?

Reply:
> no nicier way?

we don't need to use `accept` to visit the element. Not short but bit straightforward is:

```cpp
--- a/src/codegen/codegen_c_visitor.cpp
+++ b/src/codegen/codegen_c_visitor.cpp
@@ -2025,13 +2025,16 @@ void CodegenCVisitor::print_nmodl_constants() {
         printer->add_newline(2);
         printer->add_line("/** constants used in nmodl from CONSTANTS */");
         for (const auto& it: info.constants_definitions) {
-            const std::string format_string = "static const {} {{}} = "_format(
-                it->get_value()->get_node_type() == AstNodeType::DOUBLE ? "double" : "int");
-            printer->add_indent();
-            printer->add_text(fmt::format(format_string, it->get_name()->get_value()->get_value()));
-            it->get_value()->accept(*this);
-            printer->add_text(";");
-            printer->add_newline();
+            const auto& name = it->get_name()->get_node_name();
+            std::string format_string;
+            if(it->get_value()->is_double()) {
+                const auto& value = std::dynamic_pointer_cast<ast::Double>(it->get_value());
+                format_string = "static const double {} = {}"_format(name, value->get_value());
+            } else {
+                const auto& value = std::dynamic_pointer_cast<ast::Integer>(it->get_value());
+                format_string = "static const int {} = {}"_format(name, value->get_value());
+            }
+            printer->add_line(format_string);
         }
```